pkg/encoding/sanify.go

```diff
@@ -0,0 +1,49 @@
+package encoding
+
+import (
+	"github.com/pkg/errors"
+	"github.com/surajssd/kapp/pkg/spec"
+)
+
+func fixApp(app *spec.App) error {
+
+	// fix app.Services
+	for i, service := range app.Services {
+		if err := fixService(&service, app); err != nil {
+			return errors.Wrapf(err, "Unable to fix service: %v", service.Name)
+		}
+		app.Services[i] = service
+	}
+
+	// fix app.PersistentVolumes
+	for i, pVolume := range app.PersistentVolumes {
+		if err := fixPersistentVolume(&pVolume, app); err != nil {
+			return errors.Wrapf(err, "Unable to fix persistentVolume: %v", pVolume.Name)
+		}
+		app.PersistentVolumes[i] = pVolume
+	}
+
+	return nil
+}
+
+func fixService(service *spec.ServiceSpecMod, app *spec.App) error {
```

Comment:
@containscafeine It is a loop. Use it. I am pretty sure go devs have made it super optimum especially for a case like ours. Passing a struct and its element is a definite code smell. Would never like to see it in the code. The struct is also not so big that we have to think about such optimisations. Premature optimisation is the root of all evil.